app/controllers/mixins/middleware_common_mixin.rb

```diff
@@ -25,4 +34,43 @@ def clear_topology_breadcrumb
       @breadcrumbs.clear
     end
   end
+
+  def show_container(record, controller_name, display_name)
```

Comment:
why are the method names including "container" ? (here and below) 
it made sense for containers provider, but not for middleware :)